contracts/utils/Whitelist.sol

```diff
@@ -0,0 +1,72 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity >=0.6.0 <0.8.0;
+
+import "./AccessControl.sol";
+
+contract Whitelist is AccessControl {
+    using EnumerableSet for EnumerableSet.AddressSet;
+
+    event WhitelistEnabled(address sender);
+    event WhitelistDisabled(address sender);
+    event WhitelistAccountAdded(address account);
+    event WhitelistAccountRemoved(address account);
+
+    bytes32 public constant WHITELIST_ROLE = keccak256("WHITELIST_ROLE");
+
+    bool private _isWhitelistEnabled;
+    EnumerableSet.AddressSet private _whitelist;
+
+    modifier onlyWhitelistOrAdmin() {
+        _onlyWhitelistOrAdmin();
+        _;
+    }
+
+    modifier usingWhitelist() {
```

Comment:
Let's use just a function call `isWhitelisted` inside `deposit` instead of a modifier

Reply:
I looked at how they implement whitelists in Solidity: `onlyWhitelisted` modifier is used. I renamed it and it feels better now. It's hard to come up with a suitable name for a method that takes into account whether the whitelist is enabled or not, it turns out more confusing.